File: src/test/scala/org/sfsu/cs/misc/StopwordsTest.scala
package org.sfsu.cs.misc

import java.io.BufferedReader
import java.nio.charset.StandardCharsets
import java.nio.file.{Files, Paths}

/**
  * Created by rajanishivarajmaski1 on 7/8/18.
  */
class StopwordsTest {



  def compareStopWordsfileFileIO(s1: String, s2: String): Unit ={

    var reader : BufferedReader = null
    var stopWordSet1 = scala.collection.mutable.Set[String]()



  }


  def compareStopWordsFile(stopWordsFile1: String, stopWordsFile2: String) = {

    val stopWords1 = scala.io.Source.fromFile(stopWordsFile1).getLines().toSet
    val sortedList1 = stopWords1.toList.sortWith(_ > _).iterator

    val stopWords2 = scala.io.Source.fromFile(stopWordsFile2).getLines().toSet
    val sortedList2 = stopWords2.toList.sortWith(_ > _).iterator

    while (sortedList1.hasNext) {
      if (sortedList2.nonEmpty) {
        val s1 = sortedList1.next().trim
        val s2 = sortedList2.next().trim
        if (!s1.equals(s2)) {
          println(s1, s2)
        }else{
          println("equal", s1, s2)
        }
      }
    }


  }
}

object StopwordsTest{

  def main(args: Array[String]): Unit = {

    val stopWordsCompare = new StopwordsTest
    stopWordsCompare.compareStopWordsFile("/Users/rajanishivarajmaski1/University/csc895/selective-search/src/test/resources/stopwords.txt",
      "/Users/rajanishivarajmaski1/University/csc895/selective-search/20180708_145711_stopwordsReWriteToTest.txt")

  }
}

File: src/test/java/org/sfsu/cs/StopwordsTest.java
package org.sfsu.cs;

import java.io.BufferedReader;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.*;

/**
 * Created by rajanishivarajmaski1 on 7/8/18.
 */
public class StopwordsTest {

    public static void main(String[] args) throws IOException {
        StopwordsTest stopwords = new StopwordsTest();
        stopwords.fileCompare("/Users/rajanishivarajmaski1/University/csc895/selective-search/src/test/resources/stopwords.txt",
                "/Users/rajanishivarajmaski1/University/csc895/selective-search/20180708_145711_stopwordsReWriteToTest.txt");

    }
    public void fileCompare(String s1, String s2) throws IOException {
        ArrayList<String> stopWords1 = new ArrayList<>();
        try (BufferedReader reader = Files.newBufferedReader(
                Paths.get(s1),
                StandardCharsets.UTF_8)) {
            reader.lines().forEach(e -> stopWords1.add(e));
        } catch (IOException ex) {
        }
        Arrays.sort(stopWords1.toArray());
        Set<String> stopWordslist1 = new HashSet<>(stopWords1);
        List sortedList1 = new ArrayList(stopWordslist1);
        Collections.sort(sortedList1);

        Iterator<String> s1Iter = sortedList1.iterator();

        ArrayList<String> stopWords2 = new ArrayList<>();
        try (BufferedReader reader = Files.newBufferedReader(
                Paths.get(s2),
                StandardCharsets.UTF_8)) {
            reader.lines().forEach(e -> stopWords2.add(e));
        } catch (IOException ex) {
        }

        Arrays.sort(stopWords1.toArray());
        Set<String> stopWordslist2 = new HashSet<>(stopWords2);
        List sortedList2 = new ArrayList(stopWordslist2);
        Collections.sort(sortedList2);

        Iterator<String> s2Iter = sortedList2.iterator();



        while (s1Iter.hasNext()){
            System.out.println(s1Iter.next());
        }
        System.out.println("\n\n\n");
        System.out.println("second set");
        while (s2Iter.hasNext()){
            System.out.println(s2Iter.next());
        }

       /* while (s1Iter.hasNext()) {
            if (s2Iter.hasNext()) {
                String sp1 = s1Iter.next();
                String sp2 = s2Iter.next();
                if (sp1.equals(sp2)) {
                  //  System.out.println("equals: " + sp1);
                } else {
                    System.out.println("sp1: " + sp1 + "  sp2: "+ sp2);
                }
            }
        }*/

    }

}
